lib/API/JSON/v1/Settings.php

```diff
@@ -51,6 +57,16 @@ function set($settings = []) {
       ) {
         $this->onInactiveSubscribersIntervalChange();
       }
+
+      // Sync WooCommerce Customers list on settings change
+      $new_subscribe_old_woocommerce_customers = 0;
```

Comment:
This needs to be refactored. We shouldn't just keep adding stuff to this API function. We don't even write tests for it. 

I would suggest to:
1.  Get all settings from the settings controller at the beginning.
2. Run the save.
3. Get all settings from the settings controller and create a diff.
4. Create another class(es) that would have a method `onSettingsChange` and run the code there.

Reply:
I refactored it but left the code in the same file, because it seems logical for it to be there. What do you think?